pkg/snmp/traps/server_test.go

```diff
@@ -6,167 +6,31 @@
 package traps
 
 import (
-	"net"
-	"strconv"
 	"testing"
 
-	"github.com/gosnmp/gosnmp"
-	"github.com/stretchr/testify/assert"
+	"github.com/DataDog/datadog-agent/pkg/aggregator/mocksender"
 	"github.com/stretchr/testify/require"
 )
 
-var serverPort = getFreePort()
-
-func getFreePort() uint16 {
-	var port uint16
-	for i := 0; i < 5; i++ {
-		conn, err := net.ListenPacket("udp", ":0")
-		if err != nil {
-			continue
-		}
-		conn.Close()
-		port, err = parsePort(conn.LocalAddr().String())
-		if err != nil {
-			continue
-		}
-		listener, err := startSNMPTrapListener(&Config{Port: port}, nil)
-		if err != nil {
-			continue
-		}
-		listener.Close()
-		return port
-	}
-	panic("unable to find free port for starting the trap listener")
-}
-
-func parsePort(addr string) (uint16, error) {
-	_, portString, err := net.SplitHostPort(addr)
-	if err != nil {
-		return 0, err
-	}
-
-	port, err := strconv.ParseUint(portString, 10, 16)
-	if err != nil {
-		return 0, err
-	}
-	return uint16(port), nil
-}
-
-func TestServerV1GenericTrap(t *testing.T) {
```

Comment:
It seems you replaced server `StartServer()` test by `startSNMPTrapListener()` test.

Is there something preventing testing directly the server (StartServer) ?

Reply:
I think the tests here were moved to `listener_test.go` because they were only testing the "listening" part.

It's true that I'm not testing `StartServer` but `StartServer` is only starting the listener and the forwarder goroutines. I can try to add a small test for this though